lib/pending.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+var _ = require('./dash');
+var pendingRequests = [];
+var pending = module.exports = {};
+
+pending.add = function (timeoutId, request) {
+  pendingRequests.push({ timeoutId: timeoutId, request: request });
+};
+
+pending.remove = function (request) {
+  for (var i = 0; i < pendingRequests.length; i++) {
+    if (pendingRequests[i].request === request) {
+      clearTimeout(pendingRequests[i].timeoutId);
+      pendingRequests.splice(i, 1);
+      return;
+    }
+  }
+};
+
+pending.clear = function () {
+  pendingRequests.length = 0;
```

Comment:
Should be calling `clearTimeout` on any of the remaining pending requests here as well?